src/abstractdataframe/abstractdataframe.jl

```diff
@@ -645,79 +658,16 @@ julia> dropmissing(df, [:x, :y])
 ```
 
 """
-dropmissing(df::AbstractDataFrame,
-            cols::Union{Integer, Symbol, AbstractVector}=1:size(df, 2)) =
-    deleterows!(copy(df), findall(!, completecases(df, cols)))
```

Comment:
I'm not sure I like this change. Technically, a custom `AbstractDataFrame` implementation could implement `deleterows!` and automatically benefit from these definitions. If people call this on e.g. a `SubDataFrame`, they will get an error about `deleterows!`, which is relatively clear. If we want a more explicit error, we can define a `SubDataFrame` method which would print an error saying that it cannot be modified. We could also allow calling `deleterows!` on a  `SubDataFrame`, that would kind of make sense.

In terms of implementations, I also find it clearer to have the in-place methods defined at the same place as the copying methods.